lib/internal/RevHub.ts

```diff
@@ -0,0 +1,135 @@
+    setMotorChannelMode(motorChannel: number, motorMode: number, floatAtZero: boolean): Promise<void>;
+    getMotorChannelMode(motorChannel: number): Promise<{motorMode: number, floatAtZero: boolean}>
+    setMotorChannelEnable(motorChannel: number, enable: boolean): Promise<void>;
+    getMotorChannelEnable(motorChannel: number): Promise<boolean>;
+    setMotorChannelCurrentAlertLevel(motorChannel: number, currentLimit_mA: number): Promise<void>;
+    getMotorChannelCurrentAlertLevel(motorChannel: number): Promise<number>;
+    resetMotorEncoder(motorChannel: number): Promise<void>;
+    setMotorConstantPower(motorChannel: number, powerLevel: number): Promise<void>;
+    getMotorConstantPower(motorChannel: number): Promise<number>;
+    setMotorTargetVelocity(motorChannel: number, velocity_cps: number): Promise<void>;
+    getMotorTargetVelocity(motorChannel: number): Promise<number>;
+    setMotorTargetPosition(motorChannel: number, targetPosition_counts: number, targetTolerance_counts: number): Promise<void>;
+    getMotorTargetPosition(motorChannel: number): Promise<{targetPosition: number, targetTolerance: number}>;
+    getMotorAtTarget(motorChannel: number): Promise<boolean>;
+    getMotorEncoderPosition(motorChannel: number): Promise<number>;
+    setMotorPIDCoefficients(motorChannel: number, motorMode: number, pid: PIDCoefficients): Promise<void>;
+    getMotorPIDCoefficients(motorChannel: number, motorMode: number): Promise<PIDCoefficients>;
+
+    // PWM
+    setPWMConfiguration(pwmChannel: number, framePeriod: number): Promise<void>;
+    getPWMConfiguration(pwmChannel: number): Promise<number>;
+    setPWMPulseWidth(pwmChannel: number, pulseWidth: number): Promise<void>;
+    getPWMPulseWidth(pwmChannel: number): Promise<number>;
+    setPWMEnable(pwmChannel: number, enable: boolean): Promise<void>;
+    getPWMEnable(pwmChannel: number): Promise<boolean>;
+
+    // Servo
+    setServoConfiguration(servoChannel: number, framePeriod: number): Promise<void>;
+    getServoConfiguration(servoChannel: number): Promise<number>;
+    setServoPulseWidth(servoChannel: number, pulseWidth: number): Promise<void>;
+    getServoPulseWidth(servoChannel: number): Promise<number>;
+    setServoEnable(servoChannel: number, enable: boolean): Promise<void>;
+    getServoEnable(servoChannel: number): Promise<boolean>;
+}
+
+export async function openRevHub(serialNumber: string): Promise<RevHub> {
+    let serialPortPath = await getSerialPortPathForExHubSerial(serialNumber);
+
+    if(openSerialMap.get(serialPortPath) == undefined) {
+        openSerialMap.set(serialPortPath, await openSerial(serialPortPath));
+    }
+
+    let serial = openSerialMap.get(serialPortPath)!;
+
+    let parentHub = new (addon.RevHubInternal as typeof RevHubInternal)();
+
+    let discoveredModules = await addon.RevHubInternal.discoverRevHubs(serial);
+    let parentAddress = discoveredModules.parentAddress; //ToDo(Landry): Figure out what happens if no discovery found
```

Comment:
If the parent hub does not respond, either it has a firmware bug, or it got disconnected at just the wrong instance. I don't think we need to worry about it.

Reply:
Removing the log